Introduces some basic arg-parsing with getopts.

jloehel opened the pull request:
Signed-off-by: Jürgen Löhel <jloehel@suse.com>

Commit:
Introduces some basic arg-parsing with getopts.

Signed-off-by: Jürgen Löhel <jloehel@suse.com>

sndirsch commented on stsx:
```diff
@@ -28,67 +28,114 @@
 esac
 
+
+if ! [ "$watchdog_port" -eq "$watchdog_port" ] 2>/dev/null;then
```
Don't understand this line. This will always evaluate to false, right? What's the purpose behind this if-clause?

sndirsch commented on stsx:
```diff
@@ -28,67 +28,114 @@
+  error "Invalid port number"
+fi
+
+if ! [ "$watchdog_port" -ge 1 -a "$watchdog_port" -le 65535 ]; then
```
Don't understand this line. This will always evaluate to false, right? What's the purpose behind this if-clause?

sndirsch commented on stsx:
```diff
@@ -28,67 +28,114 @@
 esac
 
+
+if ! [ "$watchdog_port" -eq "$watchdog_port" ] 2>/dev/null;then
```
I don't understand this line.

tiwai replied:
It's for checking whether the given argument is an integer. 

However, don't forget that nc can accept a string service as port as well, not only the port integer number.

sndirsch replied:
Ah! Understood! But then in that case I suggest to simply remove the check.

jloehel replied:
Thank you Takashi. I didn't know that this ```nc google.de http``` is possible. I will remove it.

sndirsch reviewed (commented):
Looks good. I just don't understand this one if-clause.

Commit:
Removes the integer check for the watchdog port number.

Netcat accepts also an alias from /etc/services. It should be
possible to use an alias like http or www for the port argument.

Signed-off-by: Jürgen Löhel <jloehel@suse.com>
